### PR TITLE
Update cluster-kube-scheduler-operator

### PR DESCRIPTION
**Pull request by aveshagarwal**
I will be adding more commits to it soon.

@sjenning @mfojtik

### CODE REVIEW

**Comment by aveshagarwal** [comment]
@ravisantoshgudimetla

File: pkg/operator/starter.go [hunk 1]
```diff
@@ -34,9 +42,55 @@ func RunOperator(clientConfig *rest.Config, stopCh <-chan struct{}) error {
...
+	hasImageEnvVar := false
+	if imagePullSpecFromEnv := os.Getenv("IMAGE"); len(imagePullSpecFromEnv) > 0 {
+		hasImageEnvVar = true
+		requiredOperatorConfig.Spec.ImagePullSpec = imagePullSpecFromEnv
+	}
+
+	existing, err := client.KubeSchedulerOperatorConfigs().Get(requiredOperatorConfig.Name, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		if _, err := client.KubeSchedulerOperatorConfigs().Create(requiredOperatorConfig); err != nil {
+			panic(err)
```
Comment (sttts) [inline pkg/operator/starter.go]: many panics around here

File: pkg/operator/starter.go [hunk 2]
```diff
@@ -34,9 +42,55 @@ func RunOperator(clientConfig *rest.Config, stopCh <-chan struct{}) error {
...
+	}
+
+	if !hasImageEnvVar {
+		return
+	}
+
+	// If ImagePullSpec changed, update the existing config instance
+	if existing.Spec.ImagePullSpec != requiredOperatorConfig.Spec.ImagePullSpec {
+		existing.Spec.ImagePullSpec = requiredOperatorConfig.Spec.ImagePullSpec
+		if _, err := client.KubeSchedulerOperatorConfigs().Update(existing); err != nil {
```
Comment (sttts) [inline pkg/operator/starter.go]: retry handling needed?

Comment (sttts) [inline pkg/operator/starter.go]: retry handling needed?

File: manifests/image-references [hunk 3]
```diff
@@ -5,4 +5,8 @@ spec:
   - name: cluster-kube-scheduler-operator
     from:
       kind: "DockerImage"
-      name: "openshift/origin-cluster-kube-scheduler-operator:latest"
+      name: "docker.io/openshift/origin-cluster-kube-scheduler-operator:latest"
```
Comment (sttts) [inline manifests/image-references]: still docker.io?

**Comment by sttts** [comment]
Left some comments.

File: manifests/image-references (hunk 3 above)
Comment (aveshagarwal) [reply manifests/image-references]: did we start pushing these images to quay.io too?

File: pkg/operator/starter.go (hunk 1 above)
Comment (aveshagarwal) [reply pkg/operator/starter.go]: yes so far of operator config does not exist, that is fatal, can not proceed.

File: pkg/operator/starter.go [hunk 4]
```diff
@@ -34,9 +42,55 @@ func RunOperator(clientConfig *rest.Config, stopCh <-chan struct{}) error {
...
+		if _, err := client.KubeSchedulerOperatorConfigs().Create(requiredOperatorConfig); err != nil {
+			panic(err)
+		}
+		return
+	}
+	if err != nil {
+		panic(err)
+	}
+
+	if !hasImageEnvVar {
```
[the remark this answers is not in the document]
Comment (aveshagarwal) [reply pkg/operator/starter.go]: so far this function does not return anything.

**Comment by aveshagarwal** [comment]
I have updated this PR further to add operator status reporting to CVO.

**Pull request merged**